### PR TITLE
Improve error presentation to the SDK consumer

### PR DESCRIPTION
**Pull request by loudmouth**

### CODE REVIEW

File: Sources/Contentful/Client.swift [hunk 1]
```diff
@@ -217,7 +226,7 @@ open class Client {
     // Returns true if a rate limit error was returned by the API.
     fileprivate func didHandleRateLimitError(data: Data, response: URLResponse?, completion: ResultsHandler<Data>) -> Bool {
         if let timeUntilLimitReset = self.readRateLimitHeaderIfPresent(response: response) {
-            // At this point, We know for sure that the type returned by the API can be mapped to a `ContentfulError` instance.
+            // At this point, We know for sure that the type returned by the API can be mapped to a `APIError` instance.
```
Comment (Roblinde) [inline Sources/Contentful/Client.swift]: nitpicking "an `APIError`", right?

File: Sources/Contentful/Client.swift [hunk 2]
```diff
@@ -180,6 +180,15 @@ open class Client {
                 if self.didHandleRateLimitError(data: data, response: response, completion: completion) == true {
                     return // Exit if there was a RateLimitError.
                 }
+
+                // Use failable initialzer to optional rather than initializer that throws,
```
Comment (Roblinde) [inline Sources/Contentful/Client.swift]: nitpicking again: "initializer"

Comment (loudmouth) [reply Sources/Contentful/Client.swift]: perfect. will update!

File: Sources/Contentful/Client.swift (hunk 1 above)
Comment (loudmouth) [reply Sources/Contentful/Client.swift]: you're right

**Pull request merged**